src/sniffer.c

```diff
@@ -566,8 +566,12 @@ static int HashInit(HsHashes* hash)
         ret = wc_InitSha(&hash->hashSha);
 #endif
 #ifndef NO_MD5
-    if (ret == 0)
-        wc_InitMd5(&hash->hashMd5);
+    if (ret == 0) {
+        ret = wc_InitMd5(&hash->hashMd5);
+        if (ret != 0) {
+            return ret;
```

Comment:
This is not the pattern in the rest of the function, to return early.  Follow existing patterns.